http/src/request/validate.rs

```diff
@@ -8,7 +8,349 @@ use std::{
+    ///
+    /// This is defined in Discord's documentation, per
+    /// [Discord Docs/Action Rows][1].
+    ///
+    /// [`ActionRow`]: twilight_model::application::component::ActionRow
+    /// [1]: https://discord.com/developers/docs/interactions/message-components#action-rows
+    pub const ACTION_ROW_COMPONENT_COUNT: usize = 5;
+
+    /// Maximum number of root [`Component`]s in a message.
+    ///
+    /// This is defined in Discord's documentation, per
+    /// [Discord Docs/Action Row][1].
+    ///
+    /// [1]: https://discord.com/developers/docs/interactions/message-components#action-rows
+    pub const COMPONENT_COUNT: usize = 5;
+
+    /// Maximum length of a [`Component`] custom ID in codepoints.
+    ///
+    /// An example of a component with a custom ID is the
+    /// [`Button`][`Button::custom_id`].
+    ///
+    /// This is defined in Discord's documentation, per
+    /// [Discord Docs/Components][1].
+    ///
+    /// [`Button::custom_id`]: twilight_model::application::component::button::Button::custom_id
+    /// [1]: https://discord.com/developers/docs/interactions/message-components#component-object-component-structure
+    pub const COMPONENT_CUSTOM_ID_LENGTH: usize = 100;
+
+    /// Maximum [`Component`] label length in codepoints.
+    ///
+    /// An example of a component with a label is the [`Button`][`Button::label`].
+    ///
+    /// This is defined in Discord's documentation, per
+    /// [Discord Docs/Components][1].
+    ///
+    /// [`Button::label`]: twilight_model::application::component::button::Button::label
+    /// [1]: https://discord.com/developers/docs/interactions/message-components#component-object-component-structure
+    pub const COMPONENT_LABEL_LENGTH: usize = 80;
+
+    /// Maximum number of [`SelectMenuOption`]s that can be chosen in a
+    /// [`SelectMenu`].
+    ///
+    /// This is defined in Dicsord's documentation, per
+    /// [Discord Docs/Select Menu][1].
+    ///
+    /// [`SelectMenuOption`]: twilight_model::application::component::select_menu::SelectMenuOption
+    /// [`SelectMenu`]: twilight_model::application::component::select_menu::SelectMenu
+    /// [1]: https://discord.com/developers/docs/interactions/message-components#select-menu-object-select-menu-structure
+    pub const SELECT_MAXIMUM_VALUES_LIMIT: usize = 25;
+
+    /// Minimum number of [`SelectMenuOption`]s that can be chosen in a
+    /// [`SelectMenu`].
+    ///
+    /// This is defined in Dicsord's documentation, per
+    /// [Discord Docs/Select Menu][1].
+    ///
+    /// [`SelectMenuOption`]: twilight_model::application::component::select_menu::SelectMenuOption
+    /// [`SelectMenu`]: twilight_model::application::component::select_menu::SelectMenu
+    /// [1]: https://discord.com/developers/docs/interactions/message-components#select-menu-object-select-menu-structure
+    pub const SELECT_MAXIMUM_VALUES_REQUIREMENT: usize = 1;
+
+    /// Maximum number of [`SelectMenuOption`]s that must be chosen in a
```

Comment:
Can we maybe highlight the **can** and **must** to avoid confusion?

Reply:
We can change that after components is merged into main.

Reply:
In that case, all good